l10n_br_fiscal/models/document_fiscal_line_mixin_methods.py

```diff
@@ -292,28 +298,33 @@ def _update_fiscal_tax_ids(self, taxes):
     def _update_taxes(self):
         for line in self:
             compute_result = self._compute_taxes(line.fiscal_tax_ids)
-            line.amount_tax_included = compute_result.get("amount_included", 0.0)
-            line.amount_tax_not_included = compute_result.get(
-                "amount_not_included", 0.0
-            )
-            line.amount_tax_withholding = compute_result.get("amount_withholding", 0.0)
-            line.estimate_tax = compute_result.get("estimate_tax", 0.0)
-            line._apply_tax_fields(compute_result)
-
-    def _apply_tax_fields(self, compute_result):
+            to_update = {
+                "amount_tax_included": compute_result.get("amount_included", 0.0),
+                "amount_tax_not_included": compute_result.get(
+                    "amount_not_included", 0.0
+                ),
+                "amount_tax_withholding": compute_result.get("amount_withholding", 0.0),
+                "estimate_tax": compute_result.get("estimate_tax", 0.0),
+            }
+            to_update.update(line._prepare_tax_fields(compute_result))
+            # Update all fields at once
+            line.update(to_update)
+
+    def _prepare_tax_fields(self, compute_result):
         self.ensure_one()
         computed_taxes = compute_result.get("taxes", {})
+        tax_values = {}
         for tax in self.fiscal_tax_ids:
             computed_tax = computed_taxes.get(tax.tax_domain, {})
-            if hasattr(self, "%s_tax_id" % (tax.tax_domain,)):
-                # since v13, when line is a new record,
-                # line.fiscal_tax_ids recordset is made of
-                # NewId records with an origin pointing back to the original
-                # tax. tax.ids[0] is a way to the the single original tax back.
-                setattr(self, "%s_tax_id" % (tax.tax_domain,), tax.ids[0])
-                method = getattr(self, "_set_fields_%s" % (tax.tax_domain,))
-                if method:
-                    method(computed_tax)
+            tax_field_name = f"{tax.tax_domain}_tax_id"
+            if hasattr(self, tax_field_name):
+                tax_values[tax_field_name] = tax.ids[0]
+                method = getattr(self, f"_prepare_fields_{tax.tax_domain}", None)
```

Comment:
É mais rápido vc fazer um getattr dentro de um try, com pass do que fazer um hasatt.

Reply:
and this is again the kind of micro optimization you will not even be able to benchmark any micro second improvement in any realistic use case...

Really, there is no point in faking expertise in the projects with such useless micro optimization when @antoniospneto is proposing here optimizations that really improves the algorithmic cost...

I remember this is not the first time you come with such reviews and I think 10 years in the project was far enough time to learn that we need no such fake review thank you.

Reply:
@rvalyi eu só bati o olho rápido no PR e vou revisar com calma depois, foi apenas um comentário e não um review (o github tem as duas opções).

Todos tem tempo limitado, não vejo pq ficar explicando que vou revisar depois ou que não é uma revisão completa. O github tem essas funcionalidades pra isso.

Reply:
@antoniospneto pra ter algum ganho usando o que eu falei são necessárias muitas chamadas, não creio que sera o caso.

```
mileo@mileo-avell:~/dados/Projects$ python3
Python 3.10.12 (main, Jul 29 2024, 16:56:48) [GCC 11.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import timeit
>>> 
>>> codigo1 = """
... class A():
... 
...     b = 1
... 
... a = A()
... 
... if hasattr(a, 'b'):
...     c = getattr(a, 'b')
... #print(c)
... """
>>> 
>>> # Código 2
>>> codigo2 = """
... class A():
... 
...     b = 1
... 
... a = A()
... 
... try:
...     c = getattr(a, 'b')
... except:
...     pass
... #print(c)
... """
>>> 
>>> # Comparar o tempo de execução
>>> tempo_codigo1 = timeit.timeit(codigo1, number=1000000)
>>> tempo_codigo2 = timeit.timeit(codigo2, number=1000000)
>>> 
>>> print(f"Tempo de execução do Código 1: {tempo_codigo1} segundos")
Tempo de execução do Código 1: 4.424521369000104 segundos
>>> print(f"Tempo de execução do Código 2: {tempo_codigo2} segundos")
Tempo de execução do Código 2: 4.041749084999992 segundos
```

O ganho realmente esta onde você esta atuando, outra coisa que da pra olhar se os campos das tabelas relacionais consultadas estão com índices.

Reply:
O problema pode não estar 100% relacionado a escrita/update do cache e pode ter relação com a leitura.